packages/web-components/src/tooltip/tooltip.ts

```diff
@@ -0,0 +1,116 @@
+import { attr, FASTElement } from '@microsoft/fast-element';
+
+/**
+ * A Tooltip Custom HTML Element.
+ * @public
+ */
+export class Tooltip extends FASTElement {
+  public elementInternals = this.attachInternals();
+
+  /**
+   * Set the delay for the tooltip
+   */
+  @attr
+  public delay: number = 250;
+
+  /**
+   * Set the positioning of the tooltip
+   */
+  @attr
+  public positioning: string = 'above';
+
+  /**
+   * Reference to the tooltip element
+   * @internal
+   */
+  public tooltip!: HTMLElement;
+
+  /**
+   * The id of the anchor element for the tooltip
+   * @public
+   */
+  @attr
+  public anchor: string = '';
+
+  /**
+   * Reference to the anchor element
+   * @internal
+   */
+  private anchorElement: HTMLElement | null = null;
+
+  public constructor() {
+    super();
+    this.elementInternals.role = 'tooltip';
+  }
+
+  public connectedCallback(): void {
+    super.connectedCallback();
+
+    this.anchorElement = document.getElementById(this.anchor) || null;
```

Comment:
Should there be a check on the anchor element's focusability? like `this.anchorElement.tabIndex = this.anchorElement.tabIndex ?? 0`.

Reply:
This feels like an authoring quirk in some ways - "Don't put this on elements which aren't focusable". With that said, it's technically possible today to add the title attribute (problematic, I know) to any element. Should we constrain to focusable elements? What if the element _becomes_ focusable given a set of circumstances? If this is in the DOM and it's not focusable - does that create issues?

Reply:
Well I guess now you also have to make sure it has a plausible role, like `button` (wonder if `link` would also work), for VoiceOver to acknowledge the anchor element is focusable, and it's impossible for Tooltip to know because the anchor element could just be a custom element that use ElementInternal to define the role. So maybe leave as what Dave currently has, and use documentation to instruct authors?

Reply:
Yeah, I think there is a tension on considering whether or not where we explicitly _prevent_ usage. Some things are fairly intuitive to me where we need to couple parent/child implementation details around composability. In a case like this though, it would be like using `<fluent-button>` to navigate to a page - we wouldn't suggest that...the code doesn't explicitly prevent it though and I'm not sure it should. We should use guidance in a case like that. It seems like this is a guidance scenario potentially of, "Tooltips should only be used on focusable elements which have a valid role".

Reply:
Per discussion, we've decided against auto-setting `element.tabIndex = 0` for a couple reasons.

- It's a bit aggressive but could cause problems (e.g. a nested tabbable element like a Column Header where that's not allowed). 
- We could add `tabIndex` to provide an equivalent experience between hover/focus, but we could never provide the equivalent aural experience because generic elements (like DIV) will never have the tooltip/aria-describedby announcement. It would need to be an interactive element.

Resolved to use guidance as the primary fix here.